### PR TITLE
[v1.8.x][BACKPORT]Stablizing CI and making binaries apache compliant

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
Backport PRs #20014 #19930 #19974 #19506 #19522 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

## Testing ##
Tested on local CD pipeline identical to the one for v1.8.x: https://jenkins.mxnet-ci.amazon-ml.com/job/restricted-mxnet-cd/job/rohit_v1.8.x/

### CODE REVIEW

**Comment by access2rohit** [comment]
@samskalicky once this PR merges our binaries will be apache compliant

**Comment by access2rohit** [comment]
@samskalicky can you review ?

**Comment by samskalicky** [comment]
@leezu should we upgrade the CI in 1.x to Ubuntu 18 from 16? I thought we were only doing that for master/2.0 and later

**Comment by access2rohit** [comment]
@samskalicky its already merged in v1.x

**Comment by access2rohit** [comment]
@mxnet-bot run ci [unix-gpu]

File: cd/Jenkinsfile_release_job [hunk 1]
```diff
@@ -42,8 +42,8 @@ pipeline {
     // Using string instead of choice parameter to keep the changes to the parameters minimal to avoid
     // any disruption caused by different COMMIT_ID values chaning the job parameter configuration on
     // Jenkins.
-    string(defaultValue: "mxnet_lib", description: "Pipeline to build", name: "RELEASE_JOB_TYPE")
-    string(defaultValue: "cpu,native,cu100,cu101,cu102,cu110", description: "Comma separated list of variants", name: "MXNET_VARIANTS")
+    string(defaultValue: "mxnet_lib/static", description: "Pipeline to build", name: "RELEASE_JOB_TYPE")
```
Comment (mseth10) [inline cd/Jenkinsfile_release_job:45]: should be mxnet_lib, we removed "/static" in a previous commit https://github.com/apache/incubator-mxnet/commit/dd4661a12230584dc206b4bf233ede261fa64786#diff-dd43bbf192e508d18e340337cf5a6094e137fba710759718cfbde6cf38e27a54R45

File: cd/utils/artifact_repository.md [hunk 2]
```diff
@@ -17,7 +17,7 @@
...
+The artifact repository is an S3 bucket accessible only to restricted Jenkins nodes. It is used to store compiled MXNet artifacts that can be used by downstream CD pipelines to package the compiled libraries for different delivery channels (e.g. DockerHub, PyPI, Maven, etc.). The S3 object keys for the files being posted will be prefixed with the following distinguishing characteristics of the binary: branch, commit id, operating system, variant and dependency linking strategy (static or dynamic). For instance, s3://bucket/73b29fa90d3eac0b1fae403b7583fdd1529942dc/ubuntu16.04/cu100/static/libmxnet.so
```
Comment (mseth10) [inline cd/utils/artifact_repository.md:20]: nit: it's ubuntu18.04 now in s3 folders

File: ci/docker/Dockerfile.build.ubuntu_gpu_cu112 [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+COPY install/requirements /work/
+
+COPY install/ubuntu_core.sh /work/
+RUN /work/ubuntu_core.sh
+
+COPY install/deb_ubuntu_ccache.sh /work/
+RUN /work/deb_ubuntu_ccache.sh
+
+COPY install/ubuntu_python.sh /work/
+COPY install/requirements /work/
```
Comment (mseth10) [inline ci/docker/Dockerfile.build.ubuntu_gpu_cu112:34]: remove this, duplicate of line 25

**Review by mseth10 — COMMENTED** [review]
Left some minor comments.

**Review by mseth10 — APPROVED** [review]
LGTM, thanks for making the release ASF compliant.

**Pull request merged**